src/overlaybd/zfile/compressor.cpp

```diff
@@ -46,7 +49,7 @@ class BaseCompressor : public ICompressor {
     vector<unsigned char *> compressed_data;
```

Comment:
It seems that the content of compressed_data[] / uncompressed_data[] are generated in compress_batch() / decompress_bath, which get used in do_compress() / do_decompress(). 

I believe these 2 vectors are useless. One can use function-local arrays instead.